Plugin~/WebRTCPlugin/Context.cpp

```diff
@@ -127,19 +127,35 @@ namespace WebRTC
         audioDevice = new rtc::RefCountedObject<DummyAudioDevice>();
         nvVideoCapturerUnique = std::make_unique<NvVideoCapturer>();
         nvVideoCapturer = nvVideoCapturerUnique.get();
-        auto dummyVideoEncoderFactory = std::make_unique<DummyVideoEncoderFactory>(nvVideoCapturer);
 
-        peerConnectionFactory = webrtc::CreatePeerConnectionFactory(
-            workerThread.get(),
-            workerThread.get(),
-            signalingThread.get(),
-            audioDevice,
-            webrtc::CreateAudioEncoderFactory<webrtc::AudioEncoderOpus>(),
-            webrtc::CreateAudioDecoderFactory<webrtc::AudioDecoderOpus>(),
-            std::move(dummyVideoEncoderFactory),
-            webrtc::CreateBuiltinVideoDecoderFactory(),
-            nullptr,
-            nullptr);
+        if (!ContextManager::s_use_software_encoder)
+        {
+            auto dummyVideoEncoderFactory = std::make_unique<DummyVideoEncoderFactory>(nvVideoCapturer);
+
+            peerConnectionFactory = webrtc::CreatePeerConnectionFactory(
+                                    workerThread.get(),
+                                    workerThread.get(),
+                                    signalingThread.get(),
+                                    audioDevice,
+                                    webrtc::CreateAudioEncoderFactory<webrtc::AudioEncoderOpus>(),
+                                    webrtc::CreateAudioDecoderFactory<webrtc::AudioDecoderOpus>(),
+                                    std::move(dummyVideoEncoderFactory),
+                                    webrtc::CreateBuiltinVideoDecoderFactory(),
+                                    nullptr,
+                                    nullptr);
+        }else{
+            peerConnectionFactory = webrtc::CreatePeerConnectionFactory(
+                                    workerThread.get(),
+                                    workerThread.get(),
+                                    signalingThread.get(),
+                                    audioDevice,
+                                    webrtc::CreateAudioEncoderFactory<webrtc::AudioEncoderOpus>(),
+                                    webrtc::CreateAudioDecoderFactory<webrtc::AudioDecoderOpus>(),
+                                    webrtc::CreateBuiltinVideoEncoderFactory(),
```

Comment:
Could you please use less code to do this branching, because the only difference between software/hardware encoder is
- std::move(dummyVideoEncoderFactory)
- webrtc::CreateBuiltinVideoEncoderFactory(),

Reply:
ok.